node-packages/commons/src/tasks.js

```diff
@@ -394,7 +408,100 @@ async function createRemoveTask(removeData: Object) {
 
   switch (project.activeSystemsRemove) {
     case 'lagoon_openshiftRemove':
-      return sendToLagoonTasks('remove-openshift', removeData);
+    case 'lagoon_openshiftBuildDeploy':
+      if (type === 'branch') {
+        switch (project.branches) {
+          case undefined:
+          case null:
+            logger.debug(
+              `projectName: ${projectName}, branchName: ${branchName}, no branches defined in active system, assuming we want all of them`,
+            );
+            return sendToLagoonTasks('remove-openshift', removeData);
+          case 'true':
+            logger.debug(
+              `projectName: ${projectName}, branchName: ${branchName}, all branches active, therefore deploying`,
+            );
+            return sendToLagoonTasks('remove-openshift', removeData);
+          case 'false':
+            logger.debug(
+              `projectName: ${projectName}, branchName: ${branchName}, branch deployments disabled`,
+            );
+            throw new NoNeedToRemoveBranch('Branch deployments disabled');
+          default: {
+            logger.debug(
+              `projectName: ${projectName}, branchName: ${branchName}, regex ${
+                project.branches
+              }, testing if it matches`,
+            );
+            const branchRegex = new RegExp(project.branches);
+            if (branchRegex.test(branchName)) {
+              logger.debug(
+                `projectName: ${projectName}, branchName: ${branchName}, regex ${
+                  project.branches
+                } matched branchname, starting deploy`,
+              );
+              return sendToLagoonTasks('remove-openshift', removeData);
+            }
+            logger.debug(
+              `projectName: ${projectName}, branchName: ${branchName}, regex ${
+                project.branches
+              } did not match branchname, not deploying`,
+            );
+            throw new NoNeedToDeployBranch(
+              `configured regex '${
+                project.branches
+              }' does not match branchname '${branchName}'`,
+            );
+          }
+        }
+      } else if (type === 'pullrequest') {
```

Comment:
this is missing a `type === 'promote'` case, which means that environments of type `promote` are never deleted and therefore the CI fails. As `promote` environments are always created manually you can just assume that we can delete them like:

```
} else if (type === 'promote') {
  return sendToLagoonTasks('remove-openshift', removeData);
}
```